app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments = Comment.all
+  end
+
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+    @comments = @post.comments
+  end
+
+  def create
+      @post = Post.find(params[:post_id])
+      @comment = current_user.comments.new(comment_params)
+      @comment.post_id = params[:post_id]
```

Comment:
I would namespace post_id with the comment_params. That would avoid the need to access Post.